app/service/make_knight.py

```diff
@@ -0,0 +1,18 @@
+from app.components.knight import Knight
+from app.components.weapon import Weapon
+from app.components.armour import Armour
+from app.components.potion import Potion
+
+
+def make_knight(knight: dict) -> Knight:
+    name = knight["name"]
+    power = knight["power"]
+    hp = knight["hp"]
+    armour = [Armour(armour["part"], armour["protection"])
+              for armour in knight["armour"]]
+    weapon = Weapon(knight["weapon"]["name"], knight["weapon"]["power"])
+    potion = None
+    if knight.get("potion"):
+        potion = Potion(knight.get("potion").get("name"),
+                        knight.get("potion").get("effect"))
```

Comment:
```python
potion_data = knight.get("potion")
if potion_data:
    potion = Potion(potion_data.get("name"), potion_data.get("effect"))
```

Reply:
or you can use Assignment Expressions - https://peps.python.org/pep-0572/
The operator `:=` allows you to assign the value of `knight.get("potion")` to the variable `potion_data` directly within the `if` condition, avoiding a separate assignment line.

```python
if potion_data := knight.get("potion"):
    potion = Potion(potion_data.get("name"), potion_data.get("effect"))
```